src/analysis/special_functions/gamma.lean

```diff
@@ -0,0 +1,137 @@
+/-
+Copyright (c) 2022 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+import measure_theory.integral.exp_decay
+
+/-!
+# The Gamma function
+
+This file treats Euler's integral for the `Γ` function, `∫ x in Ioi 0, exp (-x) * x ^ (s - 1)`, for
+`s` a real or complex variable.
+
+We prove convergence of the integral for `1 ≤ s` in the real case, and `1 ≤ re s` in the complex
+case (which is non-optimal, but the optimal bound of `0 < s`, resp `0 < re s`, is harder to prove
+using the methods in the library). We also show `Γ(1) = 1`.
+
+The recurrence `Γ(s + 1) = s * Γ(s)`, holomorphy in `s`, and extension to the whole complex plane
+will be added in future pull requests.
+
+## Tags
+
+gamma
+-/
+
+noncomputable theory
+open filter interval_integral set real measure_theory
+open_locale topological_space
+
+section gamma_real
+
+/-- Asymptotic bound for the Γ function integrand. -/
+lemma gamma_integrand_is_O (s : ℝ) : asymptotics.is_O (λ x:ℝ, exp (-x) * x ^ s)
+  (λ x:ℝ, exp (-(1/2) * x)) at_top :=
+begin
+  refine asymptotics.is_o.is_O (asymptotics.is_o_of_tendsto _ _),
+  { intros x hx, exfalso, exact (exp_pos (-(1 / 2) * x)).ne' hx },
+  have : (λ (x:ℝ), exp (-x) * x ^ s / exp (-(1 / 2) * x)) = (λ (x:ℝ), exp ((1 / 2) * x) / x ^ s )⁻¹,
+  { ext1 x, dsimp,
+    have : exp (-x) = exp (-(1 / 2) * x) * exp (-(1 / 2) * x),
+    { rw ←real.exp_add, field_simp },
+    rw this,
+    have : exp (1 / 2 * x) = (exp (-(1 / 2) * x))⁻¹ := by { rw ←exp_neg, field_simp, },
+    rw this,
+    field_simp [(exp_pos (-x/2)).ne'], ring },
+  rw this,
+  exact (tendsto_exp_mul_div_rpow_at_top s (1 / 2) one_half_pos).inv_tendsto_at_top,
+end
+
+/-- Euler's integral for the `Γ` function (of a real variable `s`), defined as
+`∫ x in Ioi 0, exp (-x) * x ^ (s - 1)`.
+
+See `gamma_real_integral_convergent` for a proof of the convergence of the integral for `1 ≤ s`. -/
+def gamma_real_integral (s : ℝ) : ℝ := ∫ x in Ioi (0:ℝ), exp (-x) * x ^ (s - 1)
+
+/-- The integral defining the Γ function converges for real `s` with `1 ≤ s`.
+
+This is not optimal, but the optimal bound (convergence for `0 < s`) is hard to establish with the
+results currently in the library. -/
+lemma gamma_real_integral_convergent {s : ℝ} (h : 1 ≤ s) :
+  integrable_on (λ x:ℝ, exp (-x) * x ^ (s - 1)) (Ioi 0) :=
+begin
+  refine integrable_of_is_O_exp_neg one_half_pos _ (gamma_integrand_is_O _ ),
+  refine continuous_on_id.neg.exp.mul (continuous_on_id.rpow_const _),
+  intros x hx, right, simpa only [sub_nonneg] using h,
+end
+
+/- Most of this is just showing `∫ x in Ioi 0, exp (-x) = 1` -- maybe this should go elsewhere? -/
```

Comment:
```suggestion
```
Having a separate lemma saying that `∫ x in Ioi 0, exp (-x) = 1` would indeed make sense. The right location is not clear to me, though, so maybe just have it here for now, and we will move it if/when there are more infinite integrals like that.

Reply:
We have a file that proves integrability of this function. Does it make sense to compute `∫ x in Ici a, exp (-x)` and similar integrals in the same file?

Reply:
> We have a file that proves integrability of this function

Which file is that?

Reply:
The result I'm quoting here is from `exp_decay.lean` (added in a previous PR of mine). That doesn't seem the right place for a general table of infinite integrals. So I agree it sounds best just to leave it here for now.